pkg/github/macros.go

```diff
@@ -0,0 +1,72 @@
+package github
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type macroFunc func(string, []string) (string, error)
+
+func getMatches(macroName, rawSQL string) ([][]string, error) {
+	macroRegex := fmt.Sprintf("\\$__%s\\b(?:\\((.*?)\\))?", macroName)
+	rgx, err := regexp.Compile(macroRegex)
+	if err != nil {
+		return nil, err
+	}
+	return rgx.FindAllStringSubmatch(rawSQL, -1), nil
+}
+
+func trimAll(s []string) []string {
+	r := make([]string, len(s))
+	for i, v := range s {
+		r[i] = strings.TrimSpace(v)
+	}
+	return r
+}
+
+// InterPolateMacros interpolate macros on a given query string
+func InterPolateMacros(query string) (string, error) {
+	macros := map[string]macroFunc{
+		"multiVar": func(query string, args []string) (string, error) {
+			out := ""
+			prop := ""
+			for idx, arg := range args {
+				if idx == 0 {
+					prop = arg
+					continue
+				}
+				out = strings.Trim(fmt.Sprintf("%s %s:%s", out, prop, arg), " ")
+			}
+			if len(args) <= 1 {
```

Comment:
if we favor early returns, we would move these validations to the top before trying to loop over the args

Reply:
👍🏼 addressed in https://github.com/grafana/github-datasource/pull/162/commits/66119dfb4d970c02392590d087f63dfb1b49360f